src/components/verification/OTPInput.tsx

```diff
@@ -0,0 +1,51 @@
+import { Dispatch, SetStateAction, useMemo, useState } from 'react';
+import { TextInput } from './styles';
+
+type Props = {
+  length: number;
+  setCode: Dispatch<SetStateAction<string>>;
+  disallowSubmit: Dispatch<SetStateAction<boolean>>;
+};
+
+export default function OTPInput({ length, setCode, disallowSubmit }: Props): JSX.Element {
+  const otpValues = Array<string>(length).fill('');
```

Comment:
something bad happend here, constants in component shul be veeery rea, because reinitialisation will be with each rerender